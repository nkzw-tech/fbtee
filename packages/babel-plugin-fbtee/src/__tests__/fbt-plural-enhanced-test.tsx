import { describe, expect, it } from '@jest/globals';
import {
  jsCodeFbtCallSerializer,
  snapshotTransform,
  withFbtImportStatement,
} from './FbtTestUtil.tsx';

expect.addSnapshotSerializer(jsCodeFbtCallSerializer);

describe('Enhanced fbt:plural API with CLDR categories', () => {
  it('should support Russian plurals with few prop', () => {
    expect(
      snapshotTransform(
        withFbtImportStatement(`
          <fbt desc="Russian apple count">
            У вас
            <fbt:plural
              count={appleCount}
              few="яблока"
              many="яблок"
              name="appleCount"
              showCount="yes"
            >
              яблоко
            </fbt:plural>
          </fbt>
        `),
      ),
    ).toMatchSnapshot();
  });

  it('should support Arabic plurals with zero, two, and few props', () => {
    expect(
      snapshotTransform(
        withFbtImportStatement(`
          <fbt desc="Arabic photo count">
            <fbt:plural
              count={photoCount}
              zero="لا توجد صور"
              two="صورتان"
              few="صور"
              many="صورة"
              name="photoCount"
              showCount="yes"
            >
              صورة واحدة
            </fbt:plural>
          </fbt>
        `),
      ),
    ).toMatchSnapshot();
  });

  it('should support Hebrew plurals with two prop', () => {
    expect(
      snapshotTransform(
        withFbtImportStatement(`
          <fbt desc="Hebrew book count">
            <fbt:plural
              count={bookCount}
              two="שני ספרים"
              many="ספרים"
              name="bookCount"
              showCount="yes"
            >
              ספר אחד
            </fbt:plural>
          </fbt>
        `),
      ),
    ).toMatchSnapshot();
  });

  it('should maintain backward compatibility with just many prop', () => {
    expect(
      snapshotTransform(
        withFbtImportStatement(`
          <fbt desc="Simple English plural">
            <fbt:plural
              count={itemCount}
              many="items"
              name="itemCount"
              showCount="yes"
            >
              item
            </fbt:plural>
          </fbt>
        `),
      ),
    ).toMatchSnapshot();
  });

  it('should fallback to many when CLDR categories are not provided', () => {
    expect(
      snapshotTransform(
        withFbtImportStatement(`
          <fbt desc="Partial CLDR support">
            <fbt:plural
              count={count}
              few="некоторые элементы"
              many="много элементов"
              name="count"
              showCount="yes"
            >
              один элемент
            </fbt:plural>
          </fbt>
        `),
      ),
    ).toMatchSnapshot();
  });
});
