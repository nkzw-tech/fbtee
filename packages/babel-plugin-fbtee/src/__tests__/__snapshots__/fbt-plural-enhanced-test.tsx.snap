// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Enhanced fbt:plural API with CLDR categories should fallback to many when CLDR categories are not provided 1`] = `
import { fbt } from "fbtee";
fbt._(
  /* __FBT__ start */ {
    jsfbt: {
      m: [{ singular: true, token: "count", type: 2 }],
      t: {
        20: {
          desc: "Partial CLDR support",
          text: "{count} некоторые элементы",
        },
        "*": { desc: "Partial CLDR support", text: "{count} много элементов" },
        _1: { desc: "Partial CLDR support", text: "1 один элемент" },
      },
    },
    project: "",
  } /* __FBT__ end */,
  [fbt._plural(count, "count")]
);

`;

exports[`Enhanced fbt:plural API with CLDR categories should maintain backward compatibility with just many prop 1`] = `
import { fbt } from "fbtee";
fbt._(
  /* __FBT__ start */ {
    jsfbt: {
      m: [{ singular: true, token: "itemCount", type: 2 }],
      t: {
        "*": { desc: "Simple English plural", text: "{itemCount} items" },
        _1: { desc: "Simple English plural", text: "1 item" },
      },
    },
    project: "",
  } /* __FBT__ end */,
  [fbt._plural(itemCount, "itemCount")]
);

`;

exports[`Enhanced fbt:plural API with CLDR categories should support Arabic plurals with zero, two, and few props 1`] = `
import { fbt } from "fbtee";
fbt._(
  /* __FBT__ start */ {
    jsfbt: {
      m: [{ singular: true, token: "photoCount", type: 2 }],
      t: {
        8: { desc: "Arabic photo count", text: "{photoCount} صورتان" },
        16: { desc: "Arabic photo count", text: "{photoCount} لا توجد صور" },
        20: { desc: "Arabic photo count", text: "{photoCount} صور" },
        "*": { desc: "Arabic photo count", text: "{photoCount} صورة" },
        _1: { desc: "Arabic photo count", text: "1 صورة واحدة" },
      },
    },
    project: "",
  } /* __FBT__ end */,
  [fbt._plural(photoCount, "photoCount")]
);

`;

exports[`Enhanced fbt:plural API with CLDR categories should support Hebrew plurals with two prop 1`] = `
import { fbt } from "fbtee";
fbt._(
  /* __FBT__ start */ {
    jsfbt: {
      m: [{ singular: true, token: "bookCount", type: 2 }],
      t: {
        8: { desc: "Hebrew book count", text: "{bookCount} שני ספרים" },
        "*": { desc: "Hebrew book count", text: "{bookCount} ספרים" },
        _1: { desc: "Hebrew book count", text: "1 ספר אחד" },
      },
    },
    project: "",
  } /* __FBT__ end */,
  [fbt._plural(bookCount, "bookCount")]
);

`;

exports[`Enhanced fbt:plural API with CLDR categories should support Russian plurals with few prop 1`] = `
import { fbt } from "fbtee";
fbt._(
  /* __FBT__ start */ {
    jsfbt: {
      m: [{ singular: true, token: "appleCount", type: 2 }],
      t: {
        20: { desc: "Russian apple count", text: "У вас {appleCount} яблока" },
        "*": { desc: "Russian apple count", text: "У вас {appleCount} яблок" },
        _1: { desc: "Russian apple count", text: "У вас 1 яблоко" },
      },
    },
    project: "",
  } /* __FBT__ end */,
  [fbt._plural(appleCount, "appleCount")]
);

`;
